### PR TITLE
Remove unused models

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Supersedes https://github.com/Azure/autorest.csharp/pull/2166
Fixes https://github.com/Azure/autorest.csharp/issues/2157
Change previewed here: https://github.com/Azure/azure-sdk-for-net/pull/28388

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs [hunk 1]
```diff
@@ -173,59 +175,18 @@ private static void WriteExtensionPiece(GeneratedCodeWorkspace project, MgmtClie
 
         private static bool ShouldSkipModelGeneration(TypeProvider model)
         {
-            if (Configuration.MgmtConfiguration.NoPropertyTypeReplacement.Contains(model.Type.Name))
-                return false;
-
-            // TODO: A temporay fix for orphaned models in Resources SDK. These models are usually not directly used by ResourceData, but a descendant property of a PropertyReferenceType.
-            // Can go way after full orphan fix https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/6000
-            // The includeArmCore parameter should also be removed in FindForType() then.
+            // TODO -- The includeArmCore parameter should also be removed in FindForType() then.
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs:178]: A lot of things here are superseded by the `Remover`

**Review by m-nash — APPROVED** [review]
There is a test case which was validating one of the removed files which needs to get updated so we are still validating everything that we need to.  Other than that this and the azure-sdk-for-net changes look good

File: samples/Azure.ResourceManager.Sample/Generated/Models/DedicatedHostData.Serialization.cs [hunk 2]
```diff
@@ -89,7 +89,7 @@ internal static DedicatedHostData DeserializeDedicatedHostData(JsonElement eleme
                 }
                 if (property.NameEquals("location"))
                 {
-                    location = property.Value.GetString();
+                    location = (AzureLocation)property.Value.GetString();
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply samples/Azure.ResourceManager.Sample/Generated/Models/DedicatedHostData.Serialization.cs:92]: I will have some investigation about this change - to find a way to see if we could revert this in some ways

File: src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypePropertyChooser.cs [hunk 3]
```diff
@@ -77,31 +77,25 @@ public static bool TryGetCachedExactMatch(Schema schema, out CSharpType? result)
...
-                return GetObjectTypeProperty(originalType, _locationType, context);
+                return GetObjectTypeProperty(originalType, _locationType);
 
             if (originalType.Declaration.Name == "ResourceType" && (isString || frameworkType.Name == _resourceTypeType.Name))
-                return GetObjectTypeProperty(originalType, _resourceTypeType, context);
+                return GetObjectTypeProperty(originalType, _resourceTypeType);
 
             if (originalType.Declaration.Name == "Id" && (isString || frameworkType.Name == _resourceIdentifierType.Name))
-                return GetObjectTypeProperty(originalType, _resourceIdentifierType, context);
+                return GetObjectTypeProperty(originalType, _resourceIdentifierType);
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypePropertyChooser.cs:94]: These three lines are wrapping `ResourceType` `AzureLocation` and `ResourceIdentifier` into a `SystemObjectType` which is unnecessary and will lead to wrong serialization and deserialization method when generating resourcemanager.
Therefore here I simplified this to use replace the type using the corresponding Framework type

File: src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs [hunk 4]
```diff
@@ -57,7 +57,7 @@ private ConstructorInfo GetCtor(string attributeType)
         {
             "ResourceIdentifier" => type,
             "SystemData" => type,
-            _ => ReferenceTypes.IsMgmtReferenceType (type) ? typeof(string) : null,
+            _ => null,
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs:60]: This line is the reason that previously we have the explicit conversion - the `AzureLocation` and `ResourceType` are explicitly set to "SerializeAs: typeof(string)"

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs:60]: I also doubt if we still need these hard-coded thing - but considering that this is not the main purpose of this PR, maybe we could do that later

File: src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypes.cs [hunk 5]
```diff
@@ -1,21 +0,0 @@
...
-using Azure.Core;
-using Azure.ResourceManager.Models;
-
-namespace AutoRest.CSharp.Mgmt.Decorator
-{
-    internal static class ReferenceTypes
-    {
-        public static bool IsMgmtReferenceType(Type frameworkType)
-        {
-            return frameworkType.IsSubclassOf(typeof(ResourceIdentifier)) ||
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/Decorator/ReferenceTypes.cs:14]: This is removed because all the types are handled some place else

**Pull request merged**